backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response } from 'express';
+import { createUser } from '../services/user.service';
+import { getUserTokens, extractToken } from '../services/auth.service';
+import User from '../interfaces/User';
+import jwt from 'jsonwebtoken';
+
+export const register = async (req: Request, res: Response) => {
+  try {
+    const { userName, email, password } = req.body;
+    if (!userName || !email || !password) {
+      return res
+        .status(400)
+        .json({ message: 'Please provide all required fields' });
+    }
+    const user: User = { userName, email, password };
+    const newUser = await createUser(user);
+    res.status(201).json(newUser);
+  } catch (error: any) {
+    res.status(400).json({ message: error.message });
```

Comment:
```suggestion
   logger.error('Registration failed:', error);
   res.status(500).json({ message: 'Registration failed' });
```

Use more informative error messages and also use pino logger.

Also you should return http 500 error code instead of 400 since this is a server-side issue and not a client issue.